From d3e886e649c9d8f32f71acf21605f245788a4592 Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@hadrons.org>
Date: Fri, 1 Mar 2024 00:08:18 +0100
Subject: [PATCH libaio 09/26] Remove unused vsys_def.h

This was an old header to handle the syscalls, which is not used anymore
since commit 34c539fcb98c4397b9f04e93f8a475a9e445cf33.

Signed-off-by: Guillem Jover <guillem@hadrons.org>
---
 src/Makefile     |  2 +-
 src/compat-0_1.c |  1 -
 src/vsys_def.h   | 24 ------------------------
 3 files changed, 1 insertion(+), 26 deletions(-)
 delete mode 100644 src/vsys_def.h

diff --git a/src/Makefile b/src/Makefile
index 9a1ba51..8a9bfb4 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -39,7 +39,7 @@ libaio_srcs += compat-0_1.c
 libaio_objs := $(patsubst %.c,%.ol,$(libaio_srcs))
 libaio_sobjs := $(patsubst %.c,%.os,$(libaio_srcs))
 
-$(libaio_objs) $(libaio_sobjs): libaio.h vsys_def.h
+$(libaio_objs) $(libaio_sobjs): libaio.h
 
 %.os: %.c
 	$(CC) $(MK_CPPFLAGS) $(SO_CFLAGS) -c -o $@ $<
diff --git a/src/compat-0_1.c b/src/compat-0_1.c
index ac40fcc..459586c 100644
--- a/src/compat-0_1.c
+++ b/src/compat-0_1.c
@@ -22,7 +22,6 @@
 #include <asm/errno.h>
 
 #include "libaio.h"
-#include "vsys_def.h"
 
 #include "syscall.h"
 
diff --git a/src/vsys_def.h b/src/vsys_def.h
deleted file mode 100644
index dd711dc..0000000
--- a/src/vsys_def.h
+++ /dev/null
@@ -1,24 +0,0 @@
-/* libaio Linux async I/O interface
-   Copyright 2002 Red Hat, Inc.
-
-   This library is free software; you can redistribute it and/or
-   modify it under the terms of the GNU Lesser General Public
-   License as published by the Free Software Foundation; either
-   version 2 of the License, or (at your option) any later version.
-
-   This library is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-   Lesser General Public License for more details.
-
-   You should have received a copy of the GNU Lesser General Public
-   License along with this library; if not, write to the Free Software
-   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
- */
-extern int vsys_io_setup(unsigned nr_reqs, io_context_t *ctxp);
-extern int vsys_io_destroy(io_context_t ctx);
-extern int vsys_io_submit(io_context_t ctx, long nr, struct iocb *iocbs[]);
-extern int vsys_io_cancel(io_context_t ctx, struct iocb *iocb);
-extern int vsys_io_wait(io_context_t ctx, struct iocb *iocb, const struct timespec *when);
-extern int vsys_io_getevents(io_context_t ctx, long nr, struct io_event *events, const struct timespec *timeout);
-
-- 
2.43.0

